<?xml version="1.0" ?>
<sdf version="1.10">
  <world name="moon_world">

<physics name="default_physics" type="ode">
  <max_step_size>0.001</max_step_size>
  <real_time_update_rate>1000</real_time_update_rate>
  <ode>
    <solver>
      <iters>100</iters>
    </solver>
  </ode>
</physics>
zz

    <!-- Lunar gravity -->
    <gravity>0 0 -1.62</gravity>

    <!-- Core simulation systems -->
    <plugin filename="gz-sim-physics-system" name="gz::sim::systems::Physics"/>
    <plugin filename="gz-sim-user-commands-system" name="gz::sim::systems::UserCommands"/>
    <plugin filename="gz-sim-scene-broadcaster-system" name="gz::sim::systems::SceneBroadcaster"/>
    <plugin filename="gz-sim-imu-system"
        name="gz::sim::systems::Imu"> </plugin  >
    <plugin name="gz::sim::systems::Sensors" filename="gz-sim-sensors-system"/>

    <!-- Directional light (moon sunlight) -->
    <light type="directional" name="sun">
      <cast_shadows>true</cast_shadows>
      <pose>0 0 10 0 0 0</pose>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
    </light>

    <!-- Moon terrain model -->
    <include>
      <uri>model://moon_terrain</uri>
    </include>

    <!-- Apollo lander model -->
    <include>
      <uri>model://lander</uri>
      <pose>0 0 900 0.223599 0.223599 0.223599</pose>
      <!-- <pose>0 0 900 0.523599 0.523599 0.523599</pose>  -->
    </include>

  </world>
</sdf>
